carina-core/src/main/java/com/qaprosoft/carina/core/foundation/report/ZafiraConfigurator.java

```diff
@@ -55,35 +55,43 @@
  */
 public class ZafiraConfigurator implements IConfigurator, ITestRailManager, IQTestManager {
     private static final Logger LOGGER = Logger.getLogger(ZafiraConfigurator.class);
+    
 
     @Override
     public ConfigurationType getConfiguration() {
         ConfigurationType conf = new ConfigurationType();
         
-        String platform = Configuration.getPlatform();
-        // override platform to register correctly on Zafira based on capabilities.platform and platformName
-        R.CONFIG.put(Parameter.PLATFORM.getKey(), platform);
-        
-        String browser = Configuration.getBrowser();
-        // override browser to register correctly on Zafira based on capabilities.browserName as well
-        R.CONFIG.put(Parameter.BROWSER.getKey(), browser);
-        
+        // read all config parameter values and put into the Zafira configXML field
         for (Parameter parameter : Parameter.values()) {
             conf.getArg().add(buildArgumentType(parameter.getKey(), R.CONFIG.get(parameter.getKey())));
         }
-
-        if (R.CONFIG.containsKey(SpecialKeywords.ACTUAL_BROWSER_VERSION)) {
-            // update browser_version in returned config to register real value instead of * of matcher
-            conf.getArg().add(buildArgumentType("browser_version", R.CONFIG.get(SpecialKeywords.ACTUAL_BROWSER_VERSION)));
+        
+        // Override using actual platform, browser etc versions
+        
+        String platform = Configuration.getPlatform();
+        if (!platform.isEmpty() && !"*".equalsIgnoreCase(platform)) {
+            LOGGER.debug("Detected platform: '" + platform + "';");
+            conf.getArg().add(buildArgumentType("platform", platform));
         }
```

Comment:
Hello! Just to disscuss, I think this sentence **(!platform.isEmpty() && !"*".equalsIgnoreCase(platform))**
could be moved into a new private boolean method called selectedDefaultPlatform, in order to gain readability of the entire method. Same approach can be applied in the following if sentences. 
if (!defaultPlatformSelected) {
            LOGGER.debug("Detected platform: '" + platform + "';");
            conf.getArg().add(buildArgumentType("platform", platform));
        }
Regards, Andres!